src/debugAdapter/goDebug.ts

```diff
@@ -692,6 +761,23 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	protected launchRequest(response: DebugProtocol.LaunchResponse, args: LaunchRequestArguments): void {
+		if (!args.program) {
+			this.sendErrorResponse(response, 3000, 'Failed to continue: The program attribute is missing in the debug configuration in launch.json');
+			return;
+		}
+		this.initLaunchAttachRequest(response, args);
+	}
+
+	protected attachRequest(response: DebugProtocol.AttachResponse, args: AttachRequestArguments): void {
```

Comment:
When attaching, here, we are not expecting user to provide the `program` property, instead we are defaulting to `cwd` provided by the user which in turn defaults to `$workspacFolder`

The main use case for the `program` attribute  was to get the `localPath` which then is used to map to the remote path. Forcing this to be the `$workspaceFolder` can cause issues.

For example, say in the remote machine I have the code at `/Users/me/myproject/mysubfolder/mypackage`
And locally, this for me maps to `/Users/me2/myproject/mysubfolder/mypackage` and I have opened the folder "myproject" which makes `$workspaceFolder` resolve to `/Users/me2/myproject`

Now in the debugConfiguration, `remotePath` will point to `/Users/me/myproject/mysubfolder/mypackage`

Now the functions `toDebuggerPath` and `toLocalPath` won't work as expected

Reply:
This has been accounted for and works as expected (I've tested the remote case where Windows is the debugger and Linux is the debuggee; will test the inverse).